cmd/rookflex/cmd/mount.go

```diff
@@ -290,5 +295,66 @@ func mountCephFS(client *rpc.Client, opts *flexvolume.AttachOptions) error {
 		log(client, fmt.Sprintf("ceph filesystem %s has been attached and mounted", opts.FsName), false)
 	}
 
-	return err
+	setFSGroup(client, opts)
+	return nil
+}
+
+// setFSGroup will set the volume ownership to the fsGroup requested in the security context of the pod mounting the storage.
+// If no fsGroup is specified, does nothing.
+// If the operation fails, the error will be logged, but will not undo the mount.
+// Follows the pattern set by the k8s volume ownership as found in
+// https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/pkg/volume/volume_linux.go#L38.
+func setFSGroup(client *rpc.Client, opts *flexvolume.AttachOptions) {
+	if opts.FsGroup == "" {
+		log(client, "no fsgroup to set for the filesystem", false)
+		return
+	}
+
+	fsGroup, err := strconv.Atoi(opts.FsGroup)
+	if err != nil {
+		log(client, fmt.Sprintf("invalid fsgroup %s. %+v", opts.FsGroup, err), true)
+		return
+	}
+
+	path := path.Join(opts.MountDir, opts.Path)
+	log(client, fmt.Sprintf("setting fsgroup %d on path %s", fsGroup, path), false)
+
+	info, err := os.Stat(path)
+	if err != nil {
+		log(client, fmt.Sprintf("failed to stat path %s. %+v", path, err), true)
+		return
+	}
+
+	stat, ok := info.Sys().(*syscall.Stat_t)
+	if !ok {
+		log(client, "failed to get stat", true)
```

Comment:
I think this sounds better and more "complete":
`failed to get stat` -> `failed to get stat syscall`